WordPress/Classes/Models/AbstractPost.m

```diff
@@ -452,8 +452,17 @@ - (BOOL)isDraft
 
 - (void)publishImmediately
 {
-    self.dateModified = [NSDate date];
-    [self setDateCreated:self.dateModified];
+    // If the post was modified later than it was created
+    // then this is a post that was published at some point
+    // we will not update the dateCreated
```

Comment:
> I'm not crazy about the name either, but couldn't come up with a better one myself. Any ideas?
Maybe just publish? 🤷‍♂️  

Side note, I just realized my comment was because I was thinking about how the feature used to work, rather than how it probably should work.  I just did a sanity check in wp-admin.  A fresh draft (or draft that was never published in the past) will show "Publish Immediately" for the publish date.  But when I take a published post and change its status to a draft, it continues to show the original publish date, and there is no "Publish Immediately" option.  So there's one more thing we need to tweak in the app.  (Ugh.)